tests/test_runlib.py

```diff
@@ -642,6 +662,16 @@ def test_create_metadata_with_expected_product(self):
     self.assertEqual(list(link.signed.products.keys()), [self.test_product])
     os.remove(self.link_name)
 
+  def test_create_metadata_with_expected_product_in_target_directory(self):
+    """Test record stop records expected product in target directory """
+    in_toto_record_start(self.step_name, [], self.key)
+    in_toto_record_stop(self.step_name, [self.test_product], self.key,
+        metadata_directory=self.metadata_directory)
+    link_path = os.path.join(self.metadata_directory, self.link_name)
+    link = Metablock.load(link_path)
+    self.assertEqual(list(link.signed.products.keys()), [self.test_product])
```

Comment:
I'm wondering if we want to store the link with and without the `metadata_directory` flag and compare them in some way.